.github/workflows/integration_tests.yml

```diff
@@ -0,0 +1,91 @@
+name: Integration tests
+
+on:
+  workflow_dispatch:
+    inputs:
+      platforms:
+        description: 'CSV of Desktop, Android and/or iOS'
+        default: 'Desktop'
+        required: true
+      apis:
+        description: 'CSV of apis to build and test'
+        default: 'admob,analytics,auth,database,dynamic_links,firestore,functions,instance_id,messaging,remote_config,storage'
```

Comment:
I can't think of a good way but it is unfortunate we have to deal with a list of strings in 2 different ways (for apis and operating_systems). In an ideal world, CSV values is the most intuitive but looks like we cant avoid json representation. Some ideas that come to mind,
- Accept both as json strings but have the python script deal with json string input (not intuitive at all if we use python script on the command line directly).
- Accept both as json string and use `fromJson` and `join` functions to create a comma separated list and pass it to the python script (makes the workflow file unreadable but there is some consistency in passing arguments to the workflow file and the way its dealing with those)

Feel free to ignore if we are not planning on passing these inputs to the workflow manually often.

Reply:
We can in fact make the operating systems input a normal CSV at the expense of some added complexity. 

My reason for thinking the JSON isn't a big deal is because of the default value. If you want to manually trigger it for just  windows and mac, for example, you'll hit a button to trigger the workflow, and the field for operating systems will be pre-populated with:
["ubuntu-latest","windows-latest","macos-latest"]

At that point you just delete the pieces you don't want and end up with:
["windows-latest","macos-latest"]

Having said that, I don't feel strongly about this either way, so if we feel strongly that having simple CSVs for all inputs is worth some additional workflow complexity, then I can make that change. It would look something like this:

on:
  workflow_dispatch:
    inputs:
      operating_systems:
        default: 'ubuntu-latest,windows-latest,macos-latest'
jobs:
  prepare_matrix:
    runs-on: ubuntu-latest
    outputs:
      matrix_os: ${{ steps.set-matrix-os.outputs.matrix_os }}
    steps:
    - id: set-matrix-os
      run: echo '::set-output name=matrix_os::[${{github.event.inputs.operating_systems}}]'
  test:
    needs: prepare_matrix
    name: ${{ matrix.os }}
    runs-on: ${{ matrix.os }}
    strategy:
      fail-fast: false
      matrix: 
        os: ${{fromJson(needs.prepare_matrix.outputs.matrix_os)}}

Except we'd need to write some bash in the prepare_matrix step to convert 'a,b,c' -> '["a","b","c"]'

Reply:
Ah ok. I feel that this is not that much harder to read and my vote would be to make the interface to the workflow consistent (CSVs). Also, "dispatch" events can be sent via REST API. These inputs will be part of the payload and will be easier if the format is the same.

Reply:
I've made the change. First uses sed to replace , with "," then shoves it into ["$(value)"] with command substitution: a,b,c -> a","b",c  -> ["a",b","c"]

`OS_JSON=[\"$(echo ${{ github.event.inputs.operating_systems }} | sed 's/,/","/g')\"]`

Simplest solution in bash I could think of.

Reply:
Nice! Thanks for doing this :).